boards/xtensa/esp32/common/src/esp32_zerocross.c

```diff
@@ -0,0 +1,167 @@
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <nuttx/irq.h>
+#include <nuttx/arch.h>
+#include <arch/board/board.h>
+#include <nuttx/sensors/zerocross.h>
+
+#include "esp32_gpio.h"
+#include "hardware/esp32_gpio_sigmap.h"
+#include "esp32-wrover-kit.h"
+#include "esp32_zerocross.h"
+
+/****************************************************************************
+ * Private Function Prototypes
+ ****************************************************************************/
+
+static void zcross_enable(const struct zc_lowerhalf_s *lower,
+                          zc_interrupt_t handler, void *arg);
+
+static void zcross_disable(void);
+static int zcross_interrupt(int irq, void *context, void *arg);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* Current interrupt handler and argument */
+
+static zc_interrupt_t g_zcrosshandler;
+static void *g_zcrossarg;
+
+/* This is the zero cross lower half driver interface */
+
+static struct zc_lowerhalf_s g_zcrosslower =
+{
+    .zc_enable = zcross_enable,
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: zcross_enable
+ *
+ * Description:
+ *   Enable interrupts on the selected zero cross pin.  And empty
+ *   set will disable all interrupts.
+ *
+ ****************************************************************************/
+
+static void zcross_enable(const struct zc_lowerhalf_s *lower,
+                          zc_interrupt_t handler, void *arg)
+{
+  irqstate_t flags;
+  int irq = ESP32_PIN2IRQ(GPIO_ZERO_CROSS_IRQ);
+  int ret;
+
+  flags = enter_critical_section();
+
+  zcross_disable();
+
+  if (handler)
+    {
+      g_zcrosshandler = handler;
+      g_zcrossarg = arg;
+    }
+
+  /* Start with all interrupts disabled */
+
+  esp32_gpioirqdisable(irq);
+
+  ret = irq_attach(irq, zcross_interrupt, NULL);
+  if (ret < 0)
+    {
+      syslog(LOG_ERR, "ERROR: zcross_enable() failed: %d\n", ret);
+      leave_critical_section(flags);
+    }
+
+  esp32_gpioirqenable(irq, RISING);
+
+  leave_critical_section(flags);
+}
+
+/****************************************************************************
+ * Name: zcross_disable
+ *
+ * Description:
+ * Nullify the handler and argument.
+ *
+ ****************************************************************************/
+
+static void zcross_disable(void)
+{
+  g_zcrosshandler = NULL;
+  g_zcrossarg = NULL;
```

Comment:
You don't need to zeroed/nullify these variables because you declared it as static, by default static variables are zeroed (saved at bss segment)